ozaria/site/components/teacher-dashboard/common/SecondaryTeacherNavigation.vue

```diff
@@ -226,6 +257,18 @@ export default {
         </li>
       </ul>
     </li>
+    <li :class="{ 'modal-highlight': showCurriculumPromotion && !hackstackModalVisibility }">
```

Comment:
If other modals are added, they'll also should be checked. The [next](https://github.com/codecombat/codecombat/pull/7595) is coming soon.
I'm not sure, but maybe there's a way to check if the curriculum modal is shown instead of checking if all the others are not visible?

Reply:
our logic of showing is controlled by modal, if parent had a control over it. That would have made it easier to handle this one. 

We should at least group the condition to show modals in a mixin like we collect redis keys in backend so that we know what modals are there and what will show up to control other modals.

Reply:
should I wait for the [test as student promotion modal](https://github.com/codecombat/codecombat/pull/7595) to get merged so it would be easier to change the logic?

Reply:
can you tell me where the files are for collecting redis keys in the backend? I'm still not sure how to implment this.